src/iff.imageio/iffinput.cpp

```diff
@@ -526,210 +593,391 @@ IffInput::readimg()
+                span<uint8_t> scratch_span(scratch);
+
+                for (int c = m_header.zbuffer_bytes() - 1; c >= 0; --c) {
+                    std::vector<uint8_t> in(tw * th);
+                    span<uint8_t> in_span(in);
+
+                    // uncompress and advance span
+                    size_t used = uncompress_rle_channel(scratch_span, in_span,
+                                                         tw * th);
+                    if (used > scratch_span.size()) {
+                        errorfmt("rle read exceeds scratch buffer");
+                        return false;
+                    }
+                    scratch_span = scratch_span.subspan(used);
+
+                    // write to output buffer
+                    for (uint32_t py = ymin; py <= ymax; py++) {
+                        uint8_t* out_dy = static_cast<uint8_t*>(m_buf.data())
+                                          + (py * m_header.width)
+                                                * m_header.pixel_bytes();
+
+                        for (uint16_t px = xmin; px <= xmax; px++) {
+                            uint8_t* out_p = out_dy
+                                             + px * m_header.pixel_bytes()
+                                             + m_header.rgba_channels_bytes()
+                                             + c;
+
+                            if (in_span.empty()) {
+                                errorfmt("in span underflow");
+                                return false;
+                            }
+                            *out_p++ = in_span.front();
+                            in_span  = in_span.subspan(1);
+                        }
+                    }
+                }
+
+            } else {
+                span<uint8_t> scratch_span(scratch);
+                size_t total_pixels   = tw * th;
+                size_t expected_bytes = total_pixels * m_header.zbuffer_bytes();
+
+                if (scratch_span.size() < expected_bytes) {
+                    errorfmt(
+                        "scratch buffer too small for uncompressed zbuffer");
+                    return false;
+                }
+
+                int sy = 0;
+                for (uint16_t py = ymin; py <= ymax; py++, sy++) {
+                    uint8_t* out_dy = m_buf.data()
+                                      + (py * m_header.width)
+                                            * m_header.pixel_bytes();
+
+                    int sx = 0;
+                    for (uint16_t px = xmin; px <= xmax; px++, sx++) {
+                        size_t pixel_index  = sy * tw + sx;
+                        size_t pixel_offset = pixel_index
+                                              * m_header.zbuffer_bytes();
+
+                        if (pixel_offset + m_header.zbuffer_bytes()
+                            > scratch_span.size()) {
+                            errorfmt("in span overflow at pixel (%u, %u)", px,
```

Comment:
```suggestion
                            errorfmt("in span overflow at pixel ({}, {})", px,
```

I'm not going to mark all of them, you can do it as well as I can